### PR TITLE
Personnel and AtBScenarioFactor Code Refactor

### PR DESCRIPTION
**Pull request by Windchild292**
This is a refactor mostly relating to the Personnel tab. I decided to do this as an individual PR before I started to work on fixing and enhancing this region of the code, to make it easier to tell what is a refactor and what is a fix.

Some changes to Scenario Generation are also included, to get those refactors included in develop.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/Campaign.java [hunk 1]
```diff
@@ -6887,15 +6854,15 @@ public PartInventory getPartInventory(Part part) {
             }
             if (part.isSamePartType(p)) {
                 if (p.isPresent()) {
-                    if (p instanceof Armor) { // ProtomekArmor and BaArmor are derived from Armor
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/Campaign.java]: Revert these `ProtomekArmor` -> `ProtoMech Armor` changes. We're referring to class names here.

**Comment by Windchild292** [comment]
Fixed the issues raised during the review.

**Comment by sixlettervariables** [comment]
Approved pending conflict resolution.

**Comment by Windchild292** [comment]
Conflicts fixed

**Comment by Windchild292** [comment]
Any reason this hasn't been merged, given the approved review?

**Comment by Windchild292** [comment]
Glad to provide a reminder, was mostly wondering if there was anything else to do here as a break from working on my gradle issues.

**Pull request merged**